log/writer.go

```diff
@@ -11,16 +11,17 @@ import (
 // If the filter is nil, the writer will pass all events through.
 // The filter function is called with the module and level of the event.
 func NewFilterWriter(parent io.Writer, filter FilterFunc) io.Writer {
-	return &filterWriter{parent, filter}
+	return &filterWriter{parent: parent, filter: filter}
 }
 
 type filterWriter struct {
-	parent io.Writer
-	filter FilterFunc
+	parent        io.Writer
+	filter        FilterFunc
+	disableFilter *bool
```

Comment:
why does this need to be a pointer?

Reply:
Because it's shared between the logger and the writer and we need to set it from the logger and make it visible to the writer.

Reply:
Can we comment on these `*bool` to note that is why they are pointers?

Reply:
I could also just put a pointer to `filterWriter` in there. Maybe that would be simpler

Reply:
Is this better https://github.com/cosmos/cosmos-sdk/pull/24720/commits/052830076506bd656cb85547e8e3ddf7f1f062c1 ?